test/coordinate_unit_test.js

```diff
@@ -0,0 +1,56 @@
+/* global describe, it */
+import proj4 from 'proj4';
+import assert from 'assert';
+import Coordinates, { UNIT } from '../src/Core/Geographic/Coordinates';
+
+// Define projection that we will use (taken from https://epsg.io/3946, Proj4js section)
+proj4.defs('EPSG:3946', '+proj=lcc +lat_1=45.25 +lat_2=46.75 +lat_0=46 +lon_0=3 +x_0=1700000 +y_0=5200000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');
+
+// Asster two float number are equals, with 5 digits precision.
+function assertFloatEqual(float1, float2) {
+    assert.equal(Number(float1).toFixed(5), Number(float2).toFixed(5));
+}
+
+// Assert two coordinates obects are equals.
+function assertCoordEqual(coord1, coord2) {
+    assert.equal(coord1._internalStorageUnit, coord2._internalStorageUnit);
+    assertFloatEqual(coord1._values[0], coord2._values[0]);
+    assertFloatEqual(coord1._values[1], coord2._values[1]);
+    assertFloatEqual(coord1._values[2], coord2._values[2]);
+}
+
+describe('Coordinate conversions', function () {
+    this.timeout(10000);
+    it('Coordinate conversion from EPSG:4326 to EPSG:4978 and back to EPSG:4326', function (done) {
```

Comment:
the `it` style in mocha should read like real English. So something like:
```js
it('should correctly convert from EPSG:4326 to EPSG:4978 and back to EPSG:4326')
```
(we can do better than that probably, but you get the idea)